src/executor/stack/executor.rs

```diff
@@ -483,6 +483,57 @@ impl<'config, 'precompiles, S: StackState<'config>, P: PrecompileSet>
 		data: Vec<u8>,
 		gas_limit: u64,
 		access_list: Vec<(H160, Vec<H256>)>,
+	) -> (ExitReason, Vec<u8>) {
+		let context = Context {
+			caller,
+			address,
+			apparent_value: value,
+		};
+
+		let transfer = 	Some(Transfer {
+			source: caller,
+			target: address,
+			value,
+		});
+
+		self.transact_call_with_context(caller, address, data, gas_limit, access_list, context, transfer)
+	}
+
+	/// Execute a `DELEGATECALL` transaction with a given caller, address, value and
+	/// gas limit and data.
+	///
+	/// Takes in an additional `access_list` parameter for EIP-2930 which was
+	/// introduced in the Ethereum Berlin hard fork. If you do not wish to use
+	/// this functionality, just pass in an empty vector.
+	pub fn transact_delegatecall(
+		&mut self,
+		caller: H160,
+		address: H160,
+		value: U256,
+		data: Vec<u8>,
+		gas_limit: u64,
+		access_list: Vec<(H160, Vec<H256>)>,
+	) -> (ExitReason, Vec<u8>) {
+		let context = Context {
+			caller,
+			address: caller,
+			apparent_value: value,
+		};
+
+		let transfer = None;
+
+		self.transact_call_with_context(caller, address, data, gas_limit, access_list, context, transfer)
+	}
+
+	fn transact_call_with_context(
+		&mut self,
+		caller: H160,
+		address: H160,
+		data: Vec<u8>,
+		gas_limit: u64,
+		access_list: Vec<(H160, Vec<H256>)>,
+		context: Context,
+		transfer: Option<Transfer>,
 	) -> (ExitReason, Vec<u8>) {
 		event!(TransactCall {
```

Comment:
Change `value` to `value: context.apparent_value` in the `event!` macro. (it currently don't build with `tracing` feature).